src/libs/model/index.ts

```diff
@@ -0,0 +1,213 @@
+    const schema = require('@schemas/versions/ontouml-v1.0.schema');
+    let validator = new Ajv().compile(schema);
+    let isValid = validator(model);
+
+    // console.log('Checking validity');
+
+    if (!isValid) {
+      throw {
+        message: 'Invalid model input.',
+        errors: validator.errors,
+      };
+    }
+
+    // Enabling memoization
+    // console.log('Enabling general memoization.');
+    this.rootPackage = model;
+    this.allElements = {};
+    this.getElements = memoizee(this.getElements);
+    this.getElementById = memoizee(this.getElementById);
+
+    // Creating elements
+    // console.log('Creating elements.');
+    this.getElements().forEach((element: IElement) => {
+      this.allElements[element.id] = element;
+      this.injectFunctions(element);
+    });
+
+    // Resolving references
+    // console.log('Resolving references.');
+    this.getElements().forEach((element: IElement) => {
+      this.replaceReferences(element);
+    });
+  }
+
+  /**
+   * Function that returns an array of contained IElement objects and updates the container IReference of all elements along the way.
+   */
+  getElements(current?: IPackage): IElement[] {
+    let contents: IElement[];
+
+    if (!current) {
+      current = this.rootPackage;
+      contents = [current, ...(current.contents ? current.contents : [])];
+    } else {
+      contents = [...(current.contents ? current.contents : [])];
+    }
+
+    current.container = null;
+
+    if (!current.contents) {
+      return [];
+    }
+
+    current.contents.forEach((element: IElement) => {
+      if (element.type === OntoUMLType.PACKAGE_TYPE) {
+        contents = [...contents, ...this.getElements(element as IPackage)];
+      } else if (element.type === OntoUMLType.RELATION_TYPE) {
+        const relation = element as IRelation;
+
+        if (relation.properties) {
+          relation.properties.forEach(
+            property => (property.container = relation),
+          );
+        }
+        contents = [
+          ...contents,
+          ...(relation.properties ? relation.properties : []),
+        ];
+      } else if (element.type === OntoUMLType.CLASS_TYPE) {
+        const _class = element as IClass;
+
+        if (_class.properties) {
+          _class.properties.forEach(property => (property.container = _class));
+        }
+        if (_class.literals) {
+          _class.literals.forEach(literal => (literal.container = _class));
+        }
+        contents = [
+          ...contents,
+          ...(_class.properties ? _class.properties : []),
+          ...(_class.literals ? _class.literals : []),
+        ];
+      }
+
+      element.container = { type: current.type, id: current.id };
+    });
+
+    return contents;
+  }
+
+  /**
+   * Returns an IElement according to its id.
+   *
+   * @param elementId - Desired element's id.
+   */
+  getElementById(elementId: string): any {
+    return this.getElements().find(element => element.id === elementId);
```

Comment:
You need put the type for all elements in `map`, `find`, etc. methods. Example:

`return this.getElements().find((element: IElement) => element.id === elementId);`